source/common/quic/envoy_quic_server_stream.cc

```diff
@@ -47,7 +47,11 @@ void EnvoyQuicServerStream::encodeHeaders(const Http::ResponseHeaderMap& headers
   // This is counting not serialized bytes in the send buffer.
   local_end_stream_ = end_stream;
   SendBufferMonitor::ScopedWatermarkBufferUpdater updater(this, this);
+  IncrementalBytesSentTracker tracker(*this);
   WriteHeaders(envoyHeadersToSpdyHeaderBlock(headers), end_stream, nullptr);
+  mutableBytesMeter()->addHeaderBytesSent(tracker.incrementalBytesSent());
```

Comment:
How about making a bit more change to IncrementalBytesSentTracker to make it reference to mutableBytesMeter() and take `bool is_header` in the constructor? So that addHeaderBytesSent() and addWireBytesSent() can go into the scoped object's destructor?

Reply:
Ah, sure. Done.